### PR TITLE
test: initialize e2e testing framework based on cypress

### PR DESCRIPTION
**Pull request by liunate**
### What does this PR do
1. Initialize e2e testing framework using cypress.  
There are already some unit tests using **jest** which defaults its testing environment on `jsdom`. However `jsdom` could have some focus issues(or other glitches) that does not always work consistently with **real** browsers such as **chrome** and **firefox**.
1. Add a cypress test case based on the **defaults** demo.

### TODOs
1. **Firefox** should be working as well though the **cypress** v5 image does not work well on ci, complaining firefox is not supported 🤷‍♀️. The same docker image works perfectly with firefox at local. I may need to check that out at cypress docker image repo.
1. More cypress test cases based on demo.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -48,3 +48,20 @@ jobs:
 
       - name: Build # Tests to see if a build can succeed
         run: yarn build
+  e2e:
+    runs-on: ubuntu-latest
+    container: cypress/included:5.0.0
+    strategy:
+      matrix:
+        browser: [chrome] # cypress executable says 'Browser: 'firefox' was not found on your system or is not supported by Cypress.' seems to be defect
```
Comment (liunate) [inline .github/workflows/ci.yml:56]: `firefox` will be added if the _unsupported_ problem at the image is solved.

**Review by liunate — COMMENTED** [review]
@stefcameron This PR is in response to the discussion [#87 comment](https://github.com/focus-trap/focus-trap-react/pull/87#issuecomment-673515078)

As #87 only updates the prop type, I am not so sure if we should add test case for that simple change. Instead I think we can start the e2e test from current demo page as it proves the react wrapper works perfectly.

I also imagine there could be more demos in this repo, or maybe exactly same ones as `focus-trap` being wrapped. So e2e tests give confidence that our react wrapper does not screw things up at each `focus-trap` features.

**Comment by stefcameron** [comment]
@liunate This is exciting! Thanks for doing this. 🎉  I will try to have a look over the next few days as time permits.

File: package.json [hunk 2]
```diff
@@ -15,6 +15,8 @@
     "release": "yarn build && changeset publish",
     "test-types": "tsc index.d.ts",
     "test-unit": "jest",
+    "test-cypress": "start-server-and-test start 9966 'cypress open'",
```
Comment (liunate) [inline package.json:18]: Developer can run e2e locally with cypress interactive UI.

File: package.json [hunk 3]
```diff
@@ -67,11 +69,15 @@
     "@babel/preset-env": "^7.10.2",
     "@babel/preset-react": "^7.10.1",
     "@changesets/cli": "^2.9.1",
+    "@testing-library/cypress": "^6.0.1",
+    "@types/jquery": "^3.5.1",
```
Comment (liunate) [inline package.json:73]: Fix the missing jquery type info error while using `@testing-library/cypress`

**Comment by stefcameron** [comment]
@liunate 
> Firefox should be working as well though the cypress v5 image does not work well on ci, complaining firefox is not supported 🤷‍♀️. The same docker image works perfectly with firefox at local. I may need to check that out at cypress docker image repo.

Were you able to track this down to a bug in the image? If so, I'm thinking we should open an issue to track adding FireFox as a browser to test on our end here once the image is fixed.

**Comment by liunate** [comment]
> @liunate
> 
> > Firefox should be working as well though the cypress v5 image does not work well on ci, complaining firefox is not supported 🤷‍♀️. The same docker image works perfectly with firefox at local. I may need to check that out at cypress docker image repo.
> 
> Were you able to track this down to a bug in the image? If so, I'm thinking we should open an issue to track adding FireFox as a browser to test on our end here once the image is fixed.

That sounds a good idea. Maybe edge could be also part of the TODO items in that issue as these three(Chrome, Firefox and Edge) are the three major players in the market. _p.s. Safari could be counted one major but unfortunately cypress does not support it at this moment_

If you see this is okay, then I can create the issue along with the open issue I submitted to cypress for tracking purpose 💪

**Comment by stefcameron** [comment]
> If you see this is okay, then I can create the issue along with the open issue I submitted to cypress for tracking purpose 💪

Fantastic, please do! 🎉

**Comment by liunate** [comment]
> > If you see this is okay, then I can create the issue along with the open issue I submitted to cypress for tracking purpose 💪
> 
> Fantastic, please do! 🎉

FYI issue created for tracking #104

**Pull request merged**